### ISSUE TITLE
version of kani available via cargo doesn't check $KANI_HOME

### ISSUE THREAD

**Issue by skewballfox** [model-checking/kani#2737]
When setting `KANI_HOME` prior to running `cargo kani setup` the install location defaults to `~/.kani` rather than 

I tried this code:

```bash
set KANI_HOME $HOME/.local/kani
cargo kani setup
```
this installed to the default location.

with Kani version: kani-0.35.0. 

also after moving it manually and running `kani --version` it didn't detect the content at `$KANI_HOME`, so I suspect that the version available via `cargo install --locked kani-verifier` is older than the version where `$KANI_HOME`  was added.

**Comment by adpaco-aws** [comment]
Hi @skewballfox , thanks for opening this issue!

Would you mind giving more details about what you want to do? That is not entirely clear to me yet.

Please note that `$KANI_HOME` should not change the install location of the `kani` and `cargo-kani` binaries, but the location of the dependencies installed as part of `cargo kani setup`. To change the binaries' install location, you should use the [`cargo install` options](https://doc.rust-lang.org/cargo/commands/cargo-install.html#install-options). Then, you can set `$KANI_HOME` to change the location of `kani`/`cargo-kani` dependencies (for example, `kani-compiler`) installed as part of `cargo kani setup` and used in subsequent calls to `kani`/`cargo kani`.

Please let us know if that solves your issue. Otherwise, please provide more details so that we can reproduce the issue on our own. Thanks!
